src/lib/interceptor.ts

```diff
@@ -240,6 +246,54 @@ export class ResponseParsingInterceptor {
   }
 }
 
+export class PreventMutationsInterceptor {
+  private requestInterceptor: grpc.Requester;
+  private blankInterceptor: grpc.Requester;
+
+  constructor() {
+    this.requestInterceptor = this.buildRequester();
+    this.blankInterceptor = buildBlankInterceptor();
+  }
+
+  public intercept(options: grpc.CallOptions, nextCall: NextCall): grpc.InterceptingCall {
+    let isMutationRequest = true;
+
+    if(options?.method_definition?.path) {
+      const { path } = options.method_definition;
+      for(const prefix of NON_MUTABLE_METHOD_PREFIXES) {
+        if(path.includes(prefix)) {
+          isMutationRequest = false;
+          break;
+        }
+      }
+    }
+
+    if(isMutationRequest) {
+      return new grpc.InterceptingCall(nextCall(options), this.requestInterceptor);
+    }
+    return new grpc.InterceptingCall(nextCall(options), this.blankInterceptor);
+  }
+
+  private buildRequester(): grpc.Requester {
+    return new grpc.RequesterBuilder()
+      .withSendMessage((message: any, next: Function) => {
+        // Force validation only for requests if the setValidateOnly method exists
+        if (message?.setValidateOnly) {
+          message.setValidateOnly(true);
+          next(message);
+        // If the request doesn't support validateOnly, we attempt to clear the operations list
+        } else if (message?.clearOperationsList) {
+          message.clearOperationsList();
+          next(message);
+        } else {
+          // Otherwise, we just give up to prevent any unwanted mutations (this throws an error)
```

Comment:
We should either log or throw our own error message here to be more helpful, otherwise it might no be clear what's going on. Or even better, might it be best to just skip the actual mutation without throwing? Just console.log() and continue -- otherwise running this in dev mode might be annoying.

Reply:
Or maybe the console.logging is best left to the logging module for later, where we can turn on logging to the console in dev mode. Up to you.

Reply:
We'll leave logging for the log feature, which hopefully will be coming soon after.